### PR TITLE
Check for not empty HTTP_HOST

### PR DESCRIPTION
**Pull request by a-kolomensky**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

Example:

``` php
$client->setServerParameter('HTTP_HOST', (string) parse_url('/index.php', PHP_URL_HOST));
```

`parse_url` return empty string, and in method `getAbsoluteUri` `$currentUri` will be contain `http:///`.

Need check that `HTTP_HOST` is exist and is not empty.

### CODE REVIEW

File: src/Symfony/Component/BrowserKit/Client.php [hunk 1]
```diff
@@ -556,7 +556,7 @@ protected function getAbsoluteUri($uri)
         } else {
             $currentUri = sprintf('http%s://%s/',
                 isset($this->server['HTTPS']) ? 's' : '',
-                isset($this->server['HTTP_HOST']) ? $this->server['HTTP_HOST'] : 'localhost'
+                isset($this->server['HTTP_HOST']) && !empty($this->server['HTTP_HOST']) ? $this->server['HTTP_HOST'] : 'localhost'
```
Comment (stof) [inline src/Symfony/Component/BrowserKit/Client.php]: you can remove the ``isset`` check then, as ``!empty`` already takes care of the check

Comment (onedevlink) [inline src/Symfony/Component/BrowserKit/Client.php]: Correct

**Comment by inso** [comment]
PR disallows value '0' as hostname but if i understand correctly, according to RFC 952 and RFC 1123 this value should be allowed

**Comment by xabbuh** [comment]
you could do a check like `isset($this->server['HTTP_HOST']) && '' !== $this->server['HTTP_HOST']`

**Comment by fabpot** [comment]
The example you give is wrong. You force the HTTP_HOST to be an empty string. So, it looks more like a problem in your code rather than in Symfony.

I'm :-1: to change that without a good reason. How have you encounter this problem?

**Comment by onedevlink** [comment]
Example taken from Codeception code: https://github.com/Codeception/Codeception/blob/2.1/src/Codeception/Module/Yii2.php#L72

You are right, empty string is incorrect parameter. But i think that ```getAbsoluteUri``` not should return broken url like ```http:///``` and in this context empty string equal localhost.

**Comment by jakzal** [comment]
@onedevlink I believe this is an issue in codeception. If an invalid URL is configured, there should be an exception thrown. Otherwise localhost would be used implicitly without user being aware of a mistake he made in a config.

**Pull request closed without merging**